src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynRuleSetGenerator.cs

```diff
@@ -59,22 +56,12 @@ public RuleAction ActiveRuleAction
         /// The ruleset can be empty if there are no active rules belonging to the repo keys "vbnet", "csharpsquid" or "roslyn.*".
         /// </summary>
         /// <exception cref="AnalysisException">if required properties that should be associated with the repo key are missing.</exception>
-        public RuleSet Generate(string language, IEnumerable<SonarRule> activeRules, IEnumerable<SonarRule> inactiveRules)
+        public RuleSet Generate(string language, IEnumerable<SonarRule> rules)
         {
-            if (activeRules == null)
-            {
-                throw new ArgumentNullException(nameof(activeRules));
-            }
-            if (inactiveRules == null)
-            {
-                throw new ArgumentNullException(nameof(inactiveRules));
-            }
-            if (language == null)
-            {
-                throw new ArgumentNullException(nameof(language));
-            }
+            _ = language ?? throw new ArgumentNullException(nameof(language));
```

Comment:
I have not decided yet if I like this trick or not 😄

Reply:
The primary intention is to use `??= throw` as it's more straight forward, but we don't have C# 8 in this project (yet). So this is a fallback for now 😄

Reply:
I'm afraid we won't get rid of the .NET Framework flavor anytime soon (i guess) :)

Reply:
@mickael-caro-sonarsource You can have C# 8 language with targeting .NET Fx 4.6, but not before 5.2 release 😅 

We have it like that in the analyzer:
https://github.com/SonarSource/sonar-dotnet/blob/814b2c15a10bd83e501d3758b9625f9b1ed3cafc/analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj#L4-L8